spec/04-basic-declarations-and-definitions.md

```diff
@@ -840,6 +840,23 @@ class C extends I {
 Here, it is OK to leave out the result type of `factorial`
 in `C`, even though the method is recursive.
 
+### Tail-Recursive Call Elimination
+
+Method definitions which contain self-recursive invocations in tail position
+are optimized for stack safety. Self-invocations which are the last operation
+before returning from the method are replaced with jumps to the beginning of
+the method, much as in a while loop. Sibling-invocations are also optimized.
+
+This transform is performed automatically, without intervention by the user;
+the user can require it with the annotation, `scala.annotation.tailrec`,
```

Comment:
I think the "require it" wording is insufficiently clear and explicit. Maybe add something like "If the annotation is present but the compiler cannot perform the transform, an error is signaled", that kind of thing?

Reply:
Nth told me elsewhere, That's not an assert, it's a require. This is `require` or error. However, as this is third edit, I will try to incorporate your words. Oh, maybe "require the transform for successful compilation."

Reply:
I'm glad I put ci on last-only. This could take a while.